### DISCUSSION TITLE
Retornando no postman status 401 Unauthorized, mesmo estando logado.

### DISCUSSION THREAD

**Discussion by angelosilva-dev** [balta-io/forum#311] [Tira Dúvidas] [answered]
**Curso 2811 - Fundamentos do ASP.NET 6** 
**Aula - Configuração > [Conclusão do Módulo](https://balta.io/player/assistir/8407be55-4137-48ae-baf5-470eef47a297/b8cf74e8-2518-4af6-9e4c-a095e583dc8e)**

**Detalhe seu problema:**
Na conclusão do modulo 5 do curso de fundamentos do ASP.NET 6, após finalizar todos os passos da aula de conclusão e realizar a execução da aplicação para testa-la no postman, o mesmo está retornando o Status: 401 Unauthorized quando chamo o endpoint  "v1/accounts/upload-image".
Antes de chamar esse endpoint,  primeiramente chamo o endpoint "v1/accounts/login" para gerar o token e utiliza-lo no authorization do endpoint "v1/accounts/upload-image".
O Token é gerado com sucesso no login, porém quando executo o endpoint do upload de imagem, me retorna o status 401 unauthorized.
Já cópiei todo o projeto que está aqui no github para a minha maquina local, e fiz o teste, mas o resultado e  mesmo.

**print/evidência do erro:**
![Captura de tela 2023-04-17 202339](https://user-images.githubusercontent.com/97709774/232631734-1c09b6bf-44d4-45e8-8ccf-def4f1452958.png)

![Captura de tela 2023-04-17 201552](https://user-images.githubusercontent.com/97709774/232631756-c4a784e4-2804-4acb-ac1c-e4ed8b17d842.png)

![Captura de tela 2023-04-17 201740](https://user-images.githubusercontent.com/97709774/232631766-c7c098e2-e606-4503-9c10-69859473e743.png)

![Captura de tela 2023-04-17 202142](https://user-images.githubusercontent.com/97709774/232631777-b87dba76-7186-4cad-84eb-72a8778cfa35.png)

Já comparei com o código original da aula e não encontrei divergência.
Pelo que entendi, o login funciona, autentica normal, gera o token, porém quando chamamos o endpoint de upload de imagem, que está com o comando [Authorize], não está conseguindo verificar que o login está autenticado e com isso não entende que o usuario está autorizado.

No banco de dados, o usuario de Id 9 está com relacionado a todas as Roles Admin, User e Author.
Na classe Program.cs, já está com o comando Authentication antes do Authorization.
Preciso de um help para entender o por que isso ocorre, falta somente esse detalhe para concluir o módulo.

**Comment by andrebaltieri** [comment]
Bom dia, Angelo, como vai?

Quando você executa o código fonte do curso (https://github.com/balta-io/2811) este erro também acontece?
Você tentou debugar para ver se o conteúdo que está indo/vindo no Token realmente é o esperado? Pode ser que tenha ido alguma sujeira no payload.


Abraços,

👉 Receba os conteúdos técnicos do balta.io direto em seu E-mail<https://go.balta.io/conteudo>

--
André Baltieri
Microsoft MVP - Visual Studio & Development Technologies

CEO - balta.io
***@***.***

________________________________
De: Angelo Silva ***@***.***>
Enviado: segunda-feira, 17 de abril de 2023 20:30
Para: balta-io/forum ***@***.***>
Cc: Subscribed ***@***.***>
Assunto: [balta-io/forum] Retornando no postman status 401 Unauthorized, mesmo estando logado. (Discussion #311)


Curso 2811 - Fundamentos do ASP.NET 6
Aula - Configuração > Conclusão do Módulo<https://balta.io/player/assistir/8407be55-4137-48ae-baf5-470eef47a297/b8cf74e8-2518-4af6-9e4c-a095e583dc8e>

Detalhe seu problema:
Na conclusão do modulo 5 do curso de fundamentos do ASP.NET 6, após finalizar todos os passos da aula de conclusão e realizar a execução da plaicação para testa-la no postman, o mesmo está retornando o Status: 401 Unauthorized quando chamo o endpoint "v1/accounts/upload-image".
Antes de chamar esse endpoint, primeiramente chamo o endpoint "v1/accounts/login" para gerar o token e utiliza-lo no authorization do endpoint "v1/accounts/upload-image".
O Token é gerado com sucesso no login, porém quando executo o endpoint do upload de imagem, me retorna o status 401 unauthorized.
Já cópiei todo os projeto que está aqui no github para a minha maquina local, e fiz o teste, mas o resultado e mesmo.

print/evidência do erro:
[Captura de tela 2023-04-17 202339]<https://user-images.githubusercontent.com/97709774/232631734-1c09b6bf-44d4-45e8-8ccf-def4f1452958.png>
[Captura de tela 2023-04-17 201552]<https://user-images.githubusercontent.com/97709774/232631756-c4a784e4-2804-4acb-ac1c-e4ed8b17d842.png>
[Captura de tela 2023-04-17 201740]<https://user-images.githubusercontent.com/97709774/232631766-c7c098e2-e606-4503-9c10-69859473e743.png>
[Captura de tela 2023-04-17 202142]<https://user-images.githubusercontent.com/97709774/232631777-b87dba76-7186-4cad-84eb-72a8778cfa35.png>

Já comparei com o código original da aula e não encontrei divergência.
Pelo que entendi, o login funciona autentica, gera o token, porém quano chamamos o endpoint de upload de imagem, que está com o comando [Authorize], não está conseguindo verificar que o login está autenticado e com isso não entende que o usuario está autorizado.

No banco de dados, o usuario de Id 9 está com relacionado a todas as Roles Admin, User e Author.
Na classe Program.cs, já está com o comando Authentication antes do Authorization.
Preciso de um help para entender o por que isso ocorre, falta somente esse detalhe para concluir o módulo.

**Comment by EduardoBontorin** [comment, marked as answer]
Boa tarde Ângelo e a todos que podem vir a ter esse problema. Estava com o mesmo problema no meu programa, e o que estava ocorrendo é que eu copiei o AppSettings.json direto do git para economizar tempo e não alterei o JwtKey. Portanto o minha chave JwtKey estava diferente   meu Configuration.cs e dava o erro mencionado.

Espero ajudar alguém com essa dica
